src/app/meta-campaign/meta-campaign.component.ts

```diff
@@ -430,6 +435,18 @@ export class MetaCampaignComponent implements AfterViewChecked, OnDestroy, OnIni
 
     // Just update the public property once.
     this.tickerItems = tickerItems;
+
+    if (this.tickerUpdateTimer) {
+      clearTimeout(this.tickerUpdateTimer);
+    }
+
+    // Load data just once, but refresh e.g. time left to launch summary once
+    // per second.
+    if (isPlatformBrowser(this.platformId)) {
+      this.tickerUpdateTimer = setTimeout(() => {
+        this.setTickerParams()
```

Comment:
@NoelLH Looks good but I have a _slight_ concern that recalling _all_ of `this.setTickerParams()` every minute might be computing a lot of unnecessary stuff. In other words the only thing we need to recompute is the time left.

I also have another slight concern that this will happen every second even when it's not really noticeable, i.e. after the campaign launches when the time left will change to days, as opposed to mins and seconds, and this case covers most of the campaign time.

Reply:
@alihejazi There is very minimal logic in the rest of the method. Compared to all the other computation I expect it to be negligible.

Calling it every 1s is no different from the old approach, or from everywhere TimeLeftPipe is still used in templates.

Reply:
Ok let's try this out then